File: type-utils/src/main/java/gov/nasa/jpl/aerie/types/ExternalEventActivitySource.java
package gov.nasa.jpl.aerie.types;

public final class ExternalEventActivitySource implements ActivitySource<ExternalEvent> { // TODO, make External Event somehow
  ExternalEvent value;

  public ExternalEventActivitySource(ExternalEvent resourceName) {
    value = resourceName;
  }

  @Override
  public ExternalEvent getValue() {
    return value;
  }
}

File: procedural/timeline/src/main/kotlin/gov/nasa/ammos/aerie/procedural/timeline/payloads/reference/ExternalEventActivitySource.kt
package gov.nasa.ammos.aerie.procedural.timeline.payloads.reference

import gov.nasa.ammos.aerie.procedural.timeline.payloads.ExternalEvent
import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.AnyDirective
import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.Directive

data class ExternalEventActivitySource (
  override val v: ExternalEvent
): ActivitySource<ExternalEvent>(v)
